Pixel Art Pad - Correct PR (#1534)
hackclub/hackclub: workshops/pixel_art_pad/README.md

---
name: 'Pixel Art Pad'
description: 'Build a pixel art pad using HTML and Java'
author: '@hackyguru'
img: 'https://cloud-2qbyb4hwx.vercel.app/0pixelpad.png'
---

# Pixel Art Pad

> Most of us like pixel art. In this workshop, I'm going to show you how to make your own pixel art pad with which you can create some amazing pixel art works.

Reviewer: ```suggestion
Most of us like pixel art. In this workshop, I'm going to show you how to make your own pixel art pad with which you can create some amazing pixel arts.
```

The pixel art pad will be looking like this (You can consider modifying the look if you wish) :

[![homepage](https://cloud-2qbyb4hwx.vercel.app/0pixelpad.png)](https://pixel-creator-2.19eucs071kumara.repl.co/)

You can also take a look at the [live demo][final_live_demo] and [final code][final_code].

[final_live_demo]: https://pixel-creator-2.19eucs071kumara.repl.co/
[final_code]: https://repl.it/@19EUCS071KUMARA/Pixel-Art-Pad
[repl_it]: https://repl.it

## Part 1: Prerequisites

It is recommended to have a basic understanding of:

- HTML
- Java

> We will be using a library called processing.js which would help us in processing the graphics in a web canvas. All the functionalities of the pixel art pad will be written on java.

Reviewer: ```suggestion
We will be using a library called processing.js which will help us in processing the graphics in a web canvas. All the functionalities of the pixel art pad will be written in java.
```

However, this workshop is beginner friendly and you can refer the [final code][final_code] to understand better.

## Part 2: Getting started

### Setting up Repl.it

> [Repl.it](https://repl.it) is an amazing online code editor where we will writing our code. Though repl.it is not mandatory, I strongly recommend you to use it in order to avoid installations and downloads.

Reviewer: ```suggestion
[Repl.it](https://repl.it) is an amazing online code editor where we will be writing our code. Though repl.it is not mandatory, I strongly recommend you to use it in order to avoid installations and downloads.
```

To get started, Create your Repl by going to [https://repl.it/languages/html](https://repl.it/languages/html). Your coding environment will be created in a few moments.

![Setting up repl.it](https://cloud-flfptkrmk.vercel.app/0setuprepl.gif)

Once your repl is setup, you are all set to continue!

## Part 3: Setting Up The Required Files

### 1) Deleting the files which are not required

The first file that appears soon after the repl is setup is `index.html`.

> You will be able to find two other files called `style.css` and `script.js` in the files pane in the left side. Delete both the files as they will be required in this workshop (You can have them if you want to use custom css or js).

Reviewer: they "will be required"?

![Deleting files](https://cloud-63ot6za6a.vercel.app/0deletingfiles.gif)

### 2) Creating the java file

> We will be requiring a java file in this project. The java file will provide functionality to the web canvas. Create a file called `index.java` by pressing the new file icon on the files pane.

Reviewer: ```suggestion
We need a java file in this project. The java file will provide functionality to the web canvas. Create a file called `index.java` by pressing the new file icon on the files pane.
```

![Creating index.java](https://cloud-czzce7b1v.vercel.app/0creatingjava.gif)

Finally, you will be having two files called `index.html` and `index.java`.

## Part 4: Building The Pixel Art Pad

### 1) index.html

> Firstly, let us write the required HTML code for the pixel art pad. You will be having the following code in `index.html` by default :

Reviewer: ```suggestion
First, let us write the required HTML code for the pixel art pad. You will be having the following code in `index.html` by default :
```

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width">
    <title>repl.it</title>
    <link href="style.css" rel="stylesheet" type="text/css" />
  </head>
  <body>
    <script src="script.js"></script>
  </body>
</html>
```

Replace the text inside the title tag to the title you want. In my case, I am replacing it with 

```html 
<title>Pixel Art Pad</title>
```

If you don't use a custom CSS, remove the following line from the code  :

```html
<link href="style.css" rel="stylesheet" type="text/css" />
```

> Since we don't use a separate JavaScript file, delete the following line :

Reviewer: ```suggestion
Since we are not using JavaScript, we will also remove the following line which is inside the `<body>` tag:
```

```html
<script src="script.js"></script>
```
#### The canvas element

> Inside the `<body></body>` tag, specify the center tag followed by the heading and and `<canvas>` tag which is used for creating a canvas (for the pixel art pad).

Reviewer: grammar is a bit off in this line
